testplan/base.py

```diff
@@ -157,127 +299,14 @@ def run(self):
         return result
 
     @classmethod
-    def main_wrapper(cls,
-        name,
-        description=None,
-        parse_cmdline=True,
-        interactive=False,
-        port=None,
-        abort_signals=None,
-        logger_level=logger.TEST_INFO,
-        file_log_level=logger.DEBUG,
-        runpath=None,
-        path_cleanup=True,
-        all_tasks_local=False,
-        shuffle=None,
-        shuffle_seed=float(random.randint(1, 9999)),
-        exporters=None,
-        stdout_style=None,
-        report_dir=None,
-        xml_dir=None,
-        pdf_path=None,
-        json_path=None,
-        pdf_style=None,
-        report_tags=None,
-        report_tags_all=None,
-        merge_scheduled_parts=False,
-        browse=None,
-        ui_port=None,
-        web_server_startup_timeout=defaults.WEB_SERVER_TIMEOUT,
-        test_filter=None,
-        test_sorter=None,
-        test_lister=None,
-        verbose=None,
-        debug=None,
-        timeout=None,
-        interactive_handler=TestRunnerIHandler,
-        extra_deps=None,
-        **options
-    ):
+    def main_wrapper(cls, name, **options):
```

Comment:
If we use **options here, IDE won't prompt when user write @test_plan...

Reply:
Yes that is true. However absolutely we mustn't duplicate parameters and their defaults.

I did think about having the full parameter list and defaults on the test_plan decorator and just leaving the Testplan class as **kwargs. However this has the problem that defaults won't be set correctly if the Testplan class is used directly (as we do in a lot of tests). So I think this is the only option to avoid duplication.

Reply:
Didn't you just say 'explicit is better than implicit' in the previous comments, lol. Our current solution of unfolding **option is duplicating parameters and default from the Config to __init__, so this is unavoidable. Especially the duplication here is going to improve user experience in IDE, thus i think it's worthwhile.

On the contrary, the duplication in the other place - is totally internal, and i think that will be a good place to apply some utility functions to minimize copy & paste.

Reply:
OK, I'm just worried what will happen when e.g. we add a new param in Testplan class but forget to add it into the decorator, or change default in one but forget to change in another. But maybe you're right that this extra maintenance burden is worth it to improve user IDE experience.